### PR TITLE
First pass at fixing #340

### PR DESCRIPTION
**Pull request by apragsdale**

### CODE REVIEW

**Comment by apragsdale** [comment]
Essentially what I've done here is add a function to `load_dump.py` that iterates through the loaded data dict and checks for any `None` values, raising an error if it finds one. There may be a much more elegant way of doing this, and this solution feels a bit hacky. It does seem to work though, and if we're just forbidding `null` entries in the YAML until we make a final decision on what they mean and how they should be handled, this could be a reasonable stopgap.

Thoughts @jeromekelleher @grahamgower @molpopgen?

File: demes/load_dump.py
Comment (grahamgower) [inline demes/load_dump.py:116]: Maybe the error could be:
```suggestion
            raise ValueError(f"{key} must have a non-null value")
```
I think this is slightly less confusing in the case where someone accidentally omits a value (which might be more common than deliberately setting a value to `null`?).

**Review by grahamgower — APPROVED** [review]
LGTM. I can't think of a cleaner way to do this, and your approach is at least simple and self contained. Can you update the commit message to indicate the change before you merge this?

File: demes/load_dump.py
Comment (apragsdale) [reply demes/load_dump.py:116]: Good point - I'll change that.

**Pull request merged**
